pkgs/native_assets_builder/lib/src/file_system_cache/file_system_cache.dart

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:xxh3/xxh3.dart';
+
+import '../utils/file.dart';
+import '../utils/uri.dart';
+
+class FileSystemCache {
```

Comment:
Consider fusing API calls which are always used together, this leads to more robust and simpler APIs:

- `cache.readCacheFile` and `findOutdatedFileSystemEntity`
- `reset`, `hashFiles` and `persist`.

Also the name _cache_ implies something else, I think. 

This yields something like this as an API:

```dart
final class DependenciesHashFile {
   DependenciesHashFile({required File storage});

   Future<bool> isUpToDate();

   // Records hashes of files, returns`true` if any file has changed after `buildStart`.
   Future<bool> updateAfterBuild({required DateTime buildStart});
}
```